plugins/by-name/kulala/default.nix

```
@@ -0,0 +1,251 @@
+{ lib, ... }:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "kulala";
+  originalName = "kulala.nvim";
+  package = "kulala-nvim";
+
+  maintainers = [ lib.maintainers.BoneyPatel ];
+
+  settingsOptions = {
+    curl_path = defaultNullOpts.mkStr "curl" ''
+      Path to the cURL binary. Set this if cURL is installed in a non-standard location.
+    '';
+
+    display_mode =
+      defaultNullOpts.mkEnum
+        [
+          "split"
+          "float"
+        ]
+        "split"
```

Minor suggestion: you may prefer using `mkEnumFirstDefault` when the plugin-default is the first enum value:

```suggestion
    display_mode =
      defaultNullOpts.mkEnumFirstDefault
        [
          "split"
          "float"
        ]
```

---

Good call! I did not realize this was available. I have updated my enum types as suggested.